node/src/components/small_network.rs

```diff
@@ -860,10 +860,16 @@ where
                 NetworkInfoRequest::GetPeers { responder } => {
                     responder.respond(self.peers()).ignore()
                 }
-                NetworkInfoRequest::GetPeersInRandomOrder { responder } => {
-                    let mut peers_vec: Vec<NodeId> = self.peers().keys().cloned().collect();
-                    peers_vec.shuffle(rng);
-                    responder.respond(peers_vec).ignore()
+                NetworkInfoRequest::GetFullyConnectedPeers { responder } => {
+                    let symmetric_peers: Vec<NodeId> = self
+                        .connection_symmetries
+                        .iter()
+                        .filter_map(|(node_id, sym)| {
+                            matches!(sym, ConnectionSymmetry::Symmetric { .. }).then(|| *node_id)
+                        })
+                        .collect();
```

Comment:
Why not add a method on `enum ConnectionSymmetries` to return a collection of symmetric connections?

Reply:
There's no `ConnectionSymmetries`. The `ConnectionSymmetry` enum holds only connections for a single peer. The only method here would be something like `ConnectionSymmetries::is_symmetric`, which would contain the `matches!`. I opted against that for now, I'd add if I needed the functionality again.

Reply:
Ah, b/c we want to extract the `node_id`s that are keys in the hashmap of `connection_symmetries` in which a value is of type `ConnectionSymmetry` 😵